davinci/src/main/kotlin/com/pingidentity/davinci/collector/Collectors.kt

```diff
@@ -56,43 +51,77 @@ internal fun Collectors.request(context: FlowContext, request: Request): Request
  * This function takes a list of collectors and represents it as a JSON object. It iterates over the list of collectors,
  * adding each collector's key and value to the JSON object if the collector's value is not empty.
  *
- * @param collectors The list of collectors to represent as a JSON object.
  * @return A JSON object representing the list of collectors.
  */
 internal fun Collectors.asJson(): JsonObject {
     return buildJsonObject {
         forEach {
             when (it) {
-                is SubmitCollector -> {
-                    if (it.value.isNotEmpty()) {
-                        put("actionKey", it.key)
-                    }
-                }
-
-                is FlowCollector -> {
-                    if (it.value.isNotEmpty()) {
+                is SubmitCollector, is FlowCollector -> {
+                    if ((it as SingleValueCollector).value.isNotEmpty()) {
                         put("actionKey", it.key)
                     }
                 }
 
                 else -> {}
             }
         }
-        putJsonObject("formData") {
-            forEach {
-                when (it) {
-                    is TextCollector -> {
-                        if (it.value.isNotEmpty()) put(it.key, it.value)
+        val map = mutableMapOf<String, Any>()
+        forEach {
+            when (it) {
+                is TextCollector, is PasswordCollector -> {
+                    if ((it as SingleValueCollector).value.isNotEmpty()) {
+                        toMap(map, it.key, it.value)
                     }
+                }
 
-                    is PasswordCollector -> {
-                        if (it.value.isNotEmpty()) put(it.key, it.value)
+                is SingleSelectCollector -> {
+                    if (it.value.isNotEmpty()) {
+                        toMap(map, it.key, it.value)
                     }
+                }
 
-                    else -> {}
+                is MultiSelectCollector -> {
+                    if (it.value.isNotEmpty()) {
+                        toMap(map, it.key, it.value)
+                    }
                 }
             }
         }
+        put("formData", mapToJsonObject(map))
+    }
+}
+
+@Suppress("UNCHECKED_CAST")
+private fun toMap(map: MutableMap<String, Any>, key: String, value: Any) {
+    var currentMap = map
+
+    //TODO Remove this when the nested key is removed.
+    val keys = key.split(".")
+
+    for (i in keys.indices) {
+        val part = keys[i]
+        if (i == keys.size - 1) {
+            currentMap[part] = value
+        } else {
+            currentMap =
+                currentMap.getOrPut(part) { mutableMapOf<String, Any>() } as MutableMap<String, Any>
+        }
     }
 }
 
+@Suppress("UNCHECKED_CAST")
+fun mapToJsonObject(map: Map<String, Any>): JsonObject {
```

Comment:
Doc?

Reply:
This method will be removed after DaVinci fix the dot notation.